processor/deltatocumulativeprocessor/processor_test.go

```diff
@@ -5,308 +5,124 @@ package deltatocumulativeprocessor
-	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data/datatest/compare"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/metrics"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/testdata/random"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/testar"
 )
 
-func setup(t *testing.T, cfg *Config) (processor.Metrics, *consumertest.MetricsSink) {
-	t.Helper()
-
-	next := &consumertest.MetricsSink{}
-	if cfg == nil {
-		cfg = &Config{MaxStale: 0, MaxStreams: math.MaxInt}
-	}
-
-	proc, err := NewFactory().CreateMetrics(
-		context.Background(),
-		processortest.NewNopSettings(),
-		cfg,
-		next,
-	)
+func TestProcessor(t *testing.T) {
+	fis, err := os.ReadDir("testdata")
 	require.NoError(t, err)
 
-	return proc, next
-}
-
-// TestAccumulation verifies stream identification works correctly by writing
-// 100 random dps spread across 10 different streams.
-// Processor output is compared against a manual aggregation on a per-stream basis.
-//
-// Uses Sum datatype for testing, as we are not testing actual aggregation (see
-// internal/data for tests), but proper stream separation
-func TestAccumulation(t *testing.T) {
-	proc, sink := setup(t, nil)
-
-	sum := random.Sum()
+	for _, fi := range fis {
+		if !fi.IsDir() {
+			continue
+		}
 
-	// create 10 distinct streams
-	const N = 10
-	sbs := make([]SumBuilder, N)
-	for i := range sbs {
-		_, base := sum.Stream()
-		sbs[i] = SumBuilder{Metric: sum, base: base}
-	}
+		type Stage struct {
+			In  pmetric.Metrics `testar:"in,pmetric"`
+			Out pmetric.Metrics `testar:"out,pmetric"`
+		}
 
-	// init manual aggregation state
-	want := make(map[identity.Stream]data.Number)
-	for _, s := range sbs {
-		id := s.id(pmetric.AggregationTemporalityCumulative)
-		want[id] = s.point(0, 0, 0)
-	}
+		read := func(file string, into *Stage) error {
+			return testar.ReadFile(file, into,
+				testar.Parser("pmetric", unmarshalMetrics),
+			)
+		}
 
-	for i := 0; i < 100; i++ {
-		s := sbs[rand.Intn(N)]
+		dir := fi.Name()
+		t.Run(dir, func(t *testing.T) {
+			file := func(f string) string {
+				return filepath.Join("testdata", dir, f)
+			}
 
-		v := int64(rand.Intn(255))
-		ts := pcommon.Timestamp(i)
+			ctx := context.Background()
+			cfg := config(t, file("config.yaml"))
+			proc, sink := setup(t, cfg)
 
-		// write to processor
-		in := s.delta(s.point(0, ts, v))
-		rms := s.resourceMetrics(in)
-		err := proc.ConsumeMetrics(context.Background(), rms)
-		require.NoError(t, err)
+			stages, _ := filepath.Glob(file("*.test"))
+			for _, file := range stages {
```

Comment:
This fails in Windows, see https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/11406634691/job/31740841412

Reply:
Interesting, I just now realized that window tests [were skipped in this PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/11406170489/job/31739326774?pr=35562). [Pull requests targeting main](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/build-and-test-windows.yml#L12-L13) shouldn't skip this, right?

Reply:
I'm trying to get access to a windows machine to understand the failure. cc @sh0rez in case you beat me to it

Reply:
They are skipped unless the 'Run Windows' label was applied: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1ef15b92559b6f40a42ba32f3ac861b47271fa9d/.github/workflows/build-and-test-windows.yml#L49

If you need to, feel free to file a draft PR, ping me and I can add the 'Run Windows' label so you can use the PR for testing